### DISCUSSION TITLE
mkdocs Standalone (without any dependency to run)

### DISCUSSION THREAD

**Discussion by ricardok1** [mkdocs/mkdocs#2512] [Ideas]
Hi there

Can we have mkdocs running without python installed as a standalone library?

Thanks

**Comment by facelessuser** [comment]
I guess you could. There are a number of packages that can create an executable from Python scripts, like PyInstaller. Granted they all basically create an executable that embeds a Python installation into the binary that gets unpacked.

Going down one of these routes can be quite tricky though and may require modifications to some packaged dependencies to run properly. Basically, trying to maintain something like that can be a lot of work.

So, theoretically possible, but I doubt you are going to get a step-by-step guide to do so from here unless someone happened to go down that path already.

Personally, I don't think MkDocs should bother entertaining such an idea. If the community managed to get something up and running (ideally without having to modify any of the libraries that needed to be packed) and they wanted to document the process or provide pre-built binaries, then I think that would be fine.

**Reply by ricardok1** [reply]
My point is to include this in another project on client premisses - any alternative?

**Reply by squidfunk** [reply]
You could use Docker. Material for MkDocs provides a Docker image that comes with all dependencies preinstalled:

```
docker pull squidfunk/mkdocs-material
```

Note that this doesn’t include the default MkDocs themes. For them to work you would have to use another image.

**Reply by ricardok1** [reply]
Docker for Standalone?!?

**Reply by pawamoy** [reply]
You always rely on something, be it a vanilla Linux/Windows/Mac OS, or one with Docker available. We don't know your client environment 😕 

Regarding your question ("mkdocs running without python installed" and "any aternative"), "docker" is a valid answer. But I guess it's not a solution for you 😅
